### PR TITLE
Add integration test for audit_exam_claim_score

### PR DESCRIPTION
**Pull request by chdietz**
Last one for exam related audit tables.

### CODE REVIEW

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java [hunk 1]
```diff
@@ -1042,4 +1042,116 @@ private void compareAuditExamItemToActual(List<Map<String, Object>> examItemList
...
+
+        assertThat(originalExamId).isEqualTo(thirdExamId);
+        assertThat(countRowsInTableWhere(jdbcTemplate, "audit_exam", " exam_id = " + originalExamId)).isEqualTo(2);
+        assertThat(countRowsInTableWhere(jdbcTemplate, "audit_exam_claim_score", " exam_id = " + originalExamId)).isEqualTo(2);
+        assertThat(countRowsInTableWhere(jdbcTemplate, "exam", " id = " + originalExamId)).isEqualTo(1);
+        assertThat(countRowsInTableWhere(jdbcTemplate, "exam_claim_score", " exam_id = " + originalExamId)).isEqualTo(3);
+        compareAuditExamClaimScoreToActual(secondExamClaimScoreListToUpdate, originalExamId, "update");
+
+        // *** Fourth upsert.  Delete one, update one, leave one unchanged and add one. ***
+        // Save third exam claim scores to compare with audit
```
Comment (chdietz) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java]: Repository supports delete so it is tested even though it appears this is not used for ICA's

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java [hunk 2]
```diff
@@ -1042,4 +1042,116 @@ private void compareAuditExamItemToActual(List<Map<String, Object>> examItemList
...
+        assertThat(countRowsInTableWhere(jdbcTemplate, "exam", " id = " + originalExamId)).isEqualTo(1);
+        assertThat(countRowsInTableWhere(jdbcTemplate, "exam_claim_score", " exam_id = " + originalExamId)).isEqualTo(3);
+        compareAuditExamClaimScoreToActual(thirdExamClaimScoreListToDelete, originalExamId, "delete");
+        compareAuditExamClaimScoreToActual(thirdExamClaimScoreListToUpdate, originalExamId, "update");
+    }
+
+    private List<ExamClaim> createClaimsForAudit(Map<Integer, Integer> claimIdMap) {
+        final List<ExamClaim> claims = newArrayList();
+
+        // todo: Use realistic larger precision after fixing precision issue.
```
Comment (chdietz) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java]: Wanted to get IT tests complete.  The issue of floats affects more than audit.

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java [hunk 3]
```diff
@@ -1042,4 +1042,116 @@ private void compareAuditExamItemToActual(List<Map<String, Object>> examItemList
...
+        claimIdMap.forEach(
+                (claimId, version) -> claims.add(ExamClaim.builder()
+                        .claimId(claimId)
+                        .category(1)
+                        .scaleScore(1d + version)
+                        .scaleScoreStdErr(1d + version)
+                        .thetaScore(1d + version)
```
Comment (malaffoon) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java]: Seems like, if you're going to test the float-precision stuff, you'd want the scores to be realistic. For example the scale score should be more like `2100.0 + 123.0*version` and theta score should be `-5.0 + 2.5*version`.

Comment (chdietz) [reply exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java]: yes I agree, that is why I put a todo at line 1122.  I view that as a separate issue and needs to update all of the tests around float, exam and exam_claim_score, update and audit.  So the audit logic is in place the test logic is in place and then we do what you suggest here.

**Pull request merged**